DSCResources/MSFT_xFirewall/MSFT_xFirewall.psm1

```diff
@@ -325,154 +257,46 @@ function Test-TargetResource
-    }
-
-    if($Access)
-    {
-        $parameters["Action"] = $Access
-    }
-
-    if($RemotePort)
-    {
-        $parameters["RemotePort"] = $RemotePort
-    }
-
-    if($LocalPort)
-    {
-        $parameters["LocalPort"] = $LocalPort
-    }
-
-    if($Protocol)
-    {
-        $parameters["Protocol"] = $Protocol
-    }
-
-    if($Description)
-    {
-        $parameters["Description"] = $Description
-    }
-
-    if($ApplicationPath)
-    {
-        $parameters["Program"] = $ApplicationPath
-    }
-
-    if($Service)
-    {
-        $parameters["Service"] = $Service
-    }
-
-    Write-Verbose "SET: Invoke Set-NetFirewallRule [$Name] with splatting its parameters"
-    & $commandName @parameters
-}
-
 # Function to validate if the supplied Rule adheres to all parameters set
-function Test-RuleHasProperties
+function Test-RuleProperties
 {
     param (
         [Parameter(Mandatory)]
         $FirewallRule,
-        [String]$Name,
-        [String]$DisplayGroup,
-        [String]$State,
-        [String[]]$Profile,
-        [String]$Direction,
-        [String]$Access,
-        [String[]]$RemotePort,
-        [String[]]$LocalPort,
-        [String]$Protocol,
-        [String]$Description,
-        [String]$ApplicationPath,
-        [String]$Service
+        [String] $Name,
+        [String] $DisplayName = $Name,
+        [String] $DisplayGroup,
+        [String] $Group,
+        [String] $Enabled,
+        [String[]] $Profile,
+        [String] $Direction,
+        [String[]] $RemotePort,
+        [String[]] $LocalPort,
+        [String] $Protocol,
+        [String] $Description,
+        [String] $ApplicationPath,
+        [String] $Service
     )
 
-    $properties = Get-FirewallRuleProperty -FirewallRule $FirewallRule -Property All
+    $properties = Get-FirewallRuleProperty -FirewallRule $FirewallRule
 
     $desiredConfigurationMatch = $true
 
     if ($Name -and ($FirewallRule.Name -ne $Name))
     {
-        Write-Verbose "Test-RuleHasProperties: Name property value - $FirewallRule.Name does not match desired state - $Name"
-
-        $desiredConfigurationMatch = $false
-    }
-
-    if ($Access -and ($FirewallRule.Action -ne $Access))
-    {
-        Write-Verbose "Test-RuleHasProperties: Access property value - $($FirewallRule.Action) does not match desired state - $Access"
-
+        Write-Verbose "$($MyInvocation.MyCommand): Name property value - $FirewallRule.Name does not match desired state - $Name"
         $desiredConfigurationMatch = $false
     }
 
-    if ($State -and ($FirewallRule.Enabled.ToString() -eq ("Enabled" -ne $State)))
+    if ($Enabled -and ($FirewallRule.Enabled.ToString() -eq $Enabled))
     {
-        Write-Verbose "Test-RuleHasProperties: State property value - $FirewallRule.Enabled.ToString() does not match desired state - $State"
-
+        Write-Verbose "$($MyInvocation.MyCommand): State property value - $FirewallRule.Enabled.ToString() does not match desired state - $Enabled"
```

Comment:
I think $FirewallRule.Enabled.ToString() needs to be enclosed in parentheses - e.g. $($FirewallRule.Enabled.ToString())